### PR TITLE
Check whether the directory exists when deleting the tablespace

### PR DESCRIPTION
**Pull request by baobao0206**
If the directory of tablespace does not exist, we should got a
error on commit transaction. But error on commit transaction will
cause a panic. So the directory of tablespace should be checked
so that we can avoid panic.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/test/regress/output/gp_tablespace.source [hunk 1]
```diff
@@ -380,3 +380,12 @@ SELECT COUNT(*) FROM tblspc_otherloc_heap;
 
 DROP TABLE tblspc_otherloc_heap;
 DROP TABLESPACE testspace_otherloc;
+CREATE TABLESPACE testspace_dir_empty LOCATION '/home/gpadmin/workspace/gpdb/src/test/regress/testtablespace';
```
Comment (paul-guo-) [inline src/test/regress/output/gp_tablespace.source:383]: The answer file needs modification.

**Comment by baobao0206** [comment]
> All the `ereport(redo ? LOG : ERROR, ...)` in this function should be converted to`ereport(redo ? LOG : WARNING, ...)`. This is because `destroy_tablespace_directories` is invoked in post-commit, after writing the commit status of WAL. There is no turning back once that happens.

Yes, I agree

File: src/test/regress/output/gp_tablespace.source (hunk 1 above)
Comment (baobao0206) [reply src/test/regress/output/gp_tablespace.source:383]: Have fixed it.

**Comment by JunfengYang** [comment]
Here are two issues about the `ereport(redo ? LOG : ERROR, ...)` modification.
1. All the callers are calling `destroy_tablespace_directories(*, true /*redo*/)`.
Since the redo flag is always set to true. So no need to convert to `ereport(redo ? LOG : WARNING, ...)`.
The modification causes issue 2.

2. Since the `saved_errno` is not used anymore, a warning will be raised.
```
tablespace.c:1142:9: warning: unused variable ‘saved_errno’ [-Wunused-variable]
   int   saved_errno = errno;
```

**Comment by baobao0206** [comment]
> Here are two issues about the `ereport(redo ? LOG : ERROR, ...)` modification.
> 
> 1. All the callers are calling `destroy_tablespace_directories(*, true /*redo*/)`.
>    Since the redo flag is always set to true. So no need to convert to `ereport(redo ? LOG : WARNING, ...)` which same with the upstream code.
>    The modification causes issue 2.
> 2. Since the `saved_errno` is not used anymore, a warning will be raised.
> 
> ```
> tablespace.c:1142:9: warning: unused variable ‘saved_errno’ [-Wunused-variable]
>    int   saved_errno = errno;
> ```

Yes, you are right, I will fix it with another pr.

**Pull request merged**